dll/shellext/zipfldr/CZipFolder.hpp

```diff
@@ -351,9 +355,34 @@ class CZipFolder :
         switch (uMsg)
         {
         case DFM_MERGECONTEXTMENU:
-            DPRINT1("FIXME: Add menu items for DFM_MERGECONTEXTMENU\n");
+        {
+            QCMINFO *pqcminfo = (QCMINFO *)lParam;
+            CComPtr<IContextMenu> spContextMenu;
+            HRESULT hr = psf->QueryInterface(IID_PPV_ARG(IContextMenu, &spContextMenu));
+            if (FAILED_UNEXPECTEDLY(hr))
+                return hr;
+            hr = spContextMenu->QueryContextMenu(pqcminfo->hmenu,
+                                                 pqcminfo->indexMenu,
+                                                 pqcminfo->idCmdFirst,
+                                                 pqcminfo->idCmdLast,
+                                                 CMF_NORMAL);
+            if (FAILED_UNEXPECTEDLY(hr))
+                return hr;
+
+            pqcminfo->indexMenu += HRESULT_CODE(hr);
             return S_OK;
+        }
         case DFM_INVOKECOMMAND:
+        {
+            CComPtr<IContextMenu> spContextMenu;
+            HRESULT hr = psf->QueryInterface(IID_PPV_ARG(IContextMenu, &spContextMenu));
+            if (FAILED_UNEXPECTEDLY(hr))
+                return hr;
+
+            CMINVOKECOMMANDINFO ici = { sizeof(ici) };
+            ici.lpVerb == MAKEINTRESOURCEA(wParam);
```

Comment:
it's an ANSI string there?

Reply:
Guess so, it doesnt warn/err.

Reply:
https://docs.microsoft.com/nl-nl/windows/desktop/api/shobjidl_core/ns-shobjidl_core-_cminvokecommandinfo
`CMINVOKECOMMANDINFO` is always ANSI.

Reply:
Indeed